picard/script/__init__.py

```diff
@@ -101,3 +114,237 @@ def enabled_tagger_scripts_texts():
+            id (str): ID code for the script. Defaults to a system generated uuid.
+            last_updated (str): The UTC date and time when the script was last updated. Defaults to current date/time.
+        """
+        self.title = title if title else DEFAULT_SCRIPT_NAME
+        self.script = script
+        if id is None:
+            self._set_new_id()
+        else:
+            self.id = id
+        if last_updated is None:
+            self.update_last_updated()
+        else:
+            self.last_updated = last_updated
+
+    def _set_new_id(self):
+        """Sets the ID of the script to a new system generated uuid.
+        """
+        self.id = str(uuid.uuid4())
+
+    def get_value(self, setting):
+        """A safe way of getting the value of the specified property setting, because
+        it handles missing properties by returning None rather than raising an exception.
+
+        Args:
+            setting (str): The setting whose value to return.
+
+        Returns:
+            any: The value of the specified setting, or None if the setting was not found.
+        """
+        if hasattr(self, setting):
+            value = getattr(self, setting)
+            if isinstance(value, str):
+                value = value.strip()
+            return value
+        return None
+
+    def update_last_updated(self):
+        """Update the last updated attribute to the current UTC date and time.
+        """
+        self.last_updated = datetime.datetime.utcnow().strftime('%Y-%m-%d %H:%M:%S UTC')
+
+    def update_script_setting(self, **kwargs):
+        """Updates the value of the specified properties.
+
+        Args:
+            **kwargs: Properties to update in the form `property=value`.
+        """
+        self._update_from_dict(kwargs)
+
+    def _update_from_dict(self, settings):
+        """Updates the values of the properties based on the contents of the dictionary provided.
+        Properties in the `settings` dictionary that are not found in the script object are ignored.
+
+        Args:
+            settings (dict): The settings to update.
+        """
+        updated = False
+        for key, value in settings.items():
+            if value is not None:
+                if hasattr(self, key):
+                    if isinstance(value, str):
+                        value = value.strip()
+                    setattr(self, key, value)
+                    if key not in self._last_updated_ignore_list:
+                        updated = True
+        # Don't overwrite `last_updated` with a generated value if a last_updated value has been provided.
+        if updated and 'last_updated' not in settings:
+            self.update_last_updated()
+
+    def copy(self):
+        """Create a copy of the current script object with updated title and last updated attributes.
+        """
+        new_object = deepcopy(self)
+        new_object.update_script_setting(title=self.title + N_(" (Copy)"))
```

Comment:
From i18n point of view such concatenation will not work well.
It should be something like:

```python
new_object.update_script_setting(title=_("%s (Copy)" % self.title))
```

Reason: in other languages the suffix (here ` (Copy)`) could be placed before the title for example.